seastar/src/spawn.rs

```diff
@@ -0,0 +1,43 @@
+use crate as seastar;
+use crate::cxx_async_local_future::IntoCxxAsyncLocalFuture;
+use core::cell::Cell;
+use ffi::*;
+use std::future::Future;
+use std::rc::Rc;
+
+#[cxx::bridge]
+mod ffi {
+    #[namespace = "seastar_ffi"]
+    unsafe extern "C++" {
+        type VoidFuture = crate::cxx_async_futures::VoidFuture;
+    }
+
+    #[namespace = "seastar_ffi::spawn"]
+    unsafe extern "C++" {
+        include!("seastar/src/spawn.hh");
+
+        fn cpp_spawn(future: VoidFuture) -> VoidFuture;
+    }
+}
+
+/// Spawns a new asynchronous task, returning an `Option<Ret>`.
+///
+/// The provided future will start running in the background immediately
+/// when `spawn` is called.
+///
+/// Spawning a task enables the task to execute concurrently to other tasks.
+///
+/// This function must be called from the context of a Seastar runtime.
+pub async fn spawn<T, Ret: 'static>(future: T) -> Option<Ret>
+where
+    T: Future<Output = Ret> + 'static,
+{
+    let x: Rc<Cell<Option<Ret>>> = Default::default();
+    let xcloned = x.clone();
+    cpp_spawn(VoidFuture::infallible_local(async move {
+        xcloned.set(Some(future.await));
+    }))
+    .await
+    .ok()?;
+    Rc::try_unwrap(x).ok().unwrap().into_inner()
```

Comment:
This line is very strange. Why `Rc::try_unwrap`?

I would imagine something like this:

```rust
x.take().unwrap()
```